receipt_handler.go

```diff
@@ -0,0 +1,89 @@
+package node
+
+import (
+	"log"
+	"sync"
+)
+
+// Takes care of maintaining maps and insures that we know which interfaces are reachable where.
+type ReceiptHandler interface {
+	AddConnection(NodeAddress, ReceiptConnection)
+	AddSentPacket(p Packet, src, next NodeAddress)
+	SendReceipt(PacketReceipt)
+}
+
+type packetRecord struct {
+	destination NodeAddress
+	src         NodeAddress
+	next        NodeAddress
+	hash        PacketHash
+}
+
+type ReceiptAction interface {
+	Receipt(PacketHash)
+}
+
+type receiptImpl struct {
```

Comment:
Implementation. I'm not super happy about defining an interface and
satisfying it with a second object, since it seems redundant.

On Mon, Mar 16, 2015, 03:51 Dimitar Dimitrov <notifications@github.com>
wrote:

> In receipt_handler.go
> <https://github.com/AutoRoute/node/pull/4#discussion_r26465001>:
>
> > +	AddSentPacket(p Packet, src, next NodeAddress)
> > +	SendReceipt(PacketReceipt)
> > +}
> > +
> > +type packetRecord struct {
> > +	destination NodeAddress
> > +	src         NodeAddress
> > +	next        NodeAddress
> > +	hash        PacketHash
> > +}
> > +
> > +type ReceiptAction interface {
> > +	Receipt(PacketHash)
> > +}
> > +
> > +type receiptImpl struct {
>
> I've seen you use this style of naming a few times now, what does Impl
> mean stand for?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AutoRoute/node/pull/4/files#r26465001>.
>